src/cpp/core/include/core/http/TcpIpAsyncClientSsl.hpp

```diff
@@ -154,6 +183,56 @@ class TcpIpAsyncClientSsl
       return util::isSslShutdownError(ec);
    }
 
+#ifdef _WIN32
+   struct WindowsCertificateStore
+   {
+      WindowsCertificateStore()
+      {
+         // load certificates from important stores
+         LPCSTR stores[] = {"ROOT", "CA"};
+         for (const LPCSTR& store : stores)
+         {
+             HCERTSTORE hStore = nullptr;
+             hStore = CertOpenSystemStore(NULL, store);
+             if (!hStore)
+             {
+                LOG_ERROR_MESSAGE("Could not open certificate store");
+                return;
+             }
+
+             PCCERT_CONTEXT pContext = nullptr;
+             while (pContext = CertEnumCertificatesInStore(hStore, pContext))
+             {
+                // convert the certificate returned from the Windows store into a
+                // format that OpenSSL can understand
+                X509* x509 = nullptr;
+                x509 = d2i_X509(nullptr,
```

Comment:
One thing that's not clear to me -- does this need to be aware of how the certificate is encoded? (that is, `dwCertEncodingType`).

Reply:
Also from https://linux.die.net/man/3/d2i_x509:

> If the call is successful *in is incremented to the byte following the parsed data.

Given that it tries to mutate this argument I wonder if we should copy it (`pbCertEncoded`) to our own temporary variable just to be safe? This also gives us an opportunity to case it as appropriate.

Reply:
@kevinushey The windows cert is always in an ASN encoding, so as far as I can tell we do not need to worry about `dwCertEncodingType`.